### PR TITLE
Add custom status codes

### LINKED ISSUE #41: Errors - Cleanup
`KappaErrorCodes.InvalidRoom` is no longer used. The class name can be renamed to incorporate status codes and moved out of `errors.py`. Rename could possibly be `KappaWebsocketCodes`.

```py
class KappaWebsocketCodes(IntEnum):
    """Custom websocket codes for returning client data."""

    KappaSuccess = 4000

    RoomNotFound = 4001
```

### PR DESCRIPTION
**Pull request by kappq**
This edits the current `errors.py` file adding changing the `KappaCloseCodes` class to `StatusCode` and adding a success code.

This also sets the code of the `RoomNotFound` error to `StatusCode.ROOM_NOT_FOUND`.

### CODE REVIEW

File: server/codes.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from enum import IntEnum
+
+
+class StatusCode(IntEnum):
+    """Custom status codes for a WebSocket event."""
+
+    SUCCESS = 200
```
Comment (LukeHankey) [inline server/codes.py]: 200 is a HTTP success code.

File: server/codes.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    INVALID_ERROR = 4000
```
Comment (LukeHankey) [inline server/codes.py]: Do we need this anymore?

File: server/codes.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class RoomNotFoundError(Exception):
+    """Custom exception raised when joining a room with an invalid code."""
+
+    def __init__(self, message: str) -> None:
+        super().__init__(message)
+
+        self.code = StatusCode.ROOM_NOT_FOUND
+        self.message = message
```
Comment (LukeHankey) [inline server/codes.py]: Should errors not be separated into their own file?

**Review by LukeHankey — CHANGES_REQUESTED** [review]

File: server/codes.py (hunk 3 above)
Comment (kappq) [reply server/codes.py]: Will do.

File: server/codes.py (hunk 2 above)
Comment (kappq) [reply server/codes.py]: Good point. Removing it.

File: server/main.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
 from fastapi import FastAPI, WebSocket, WebSocketDisconnect
 from pydantic import BaseModel
 
-from .codes import RoomNotFoundError, StatusCode
+from .errors import RoomNotFoundError, StatusCode
```
Comment (LukeHankey) [inline server/main.py]: `StatusCode` should still come from `.codes` not from `.errors`

Comment (kappq) [reply server/main.py]: Good catch! I didn't notice it because in `errors.py` I was importing `StatusCode`, so it didn't raise an `ImportError`.

**Review by LukeHankey — APPROVED** [review]

**Pull request merged**